pwa/src/components/applications/applicationForm.tsx

```diff
@@ -0,0 +1,233 @@
+import * as React from "react";
+import {
+  GenericInputComponent,
+  Spinner,
+  Card,
+  Alert,
+  Accordion
+} from "@conductionnl/nl-design-system/lib";
+import { isLoggedIn } from "../../services/auth";
+import { Link } from "gatsby";
+import { navigate } from "gatsby-link";
+import {
+  checkValues,
+  removeEmptyObjectValues, retrieveFormArrayAsOArray,
+} from "../utility/inputHandler";
+import {ArrayInputComponent} from "../common/arrayInput";
+import FlashMessage from 'react-flash-message';
+
+export default function ApplicationForm({ id }) {
+  const [context, setContext] = React.useState(null);
+  const [alert, setAlert] = React.useState(null);
+  const [application, setApplication] = React.useState(null);
+  const [showSpinner, setShowSpinner] = React.useState(false);
+
+  React.useEffect(() => {
+    if (typeof window !== "undefined" && context === null) {
+      setContext({
+        adminUrl: window.GATSBY_ADMIN_URL,
+        frontendUrl: window.GATSBY_FRONTEND_URL,
+      });
+    } else {
+      if (isLoggedIn()) {
+        if (id !== "new") {
+          getApplication();
+        }
+      }
+    }
+  }, [context]);
```

Comment:
Ik denk dat dit onnodig complex is gemaakt. Voor zover ik zie zijn de values `adminUrl` en `frontendUrl` statisch, wat inhoudt dat ze niet in een `useEffect` aangemaakt hoeven te worden.

Volgens mij zou onderstaand precies hetzelfde doen:

```
const context = {
    adminUrl: window.GATSBY_ADMIN_URL,
    frontendUrl: window.GATSBY_FRONTEND_URL,
  }

  React.useEffect(() => {
    if (isLoggedIn && id !== "new") {
      getApplication()
    }
  });
```

Daarnaast zouden we denk ik de `GATSBY_ADMIN_URL` en `FRONTEND_URL` in een `.env` kunnen bewaren waardoor we er helemaal geen variabelen voor hoeven aan te maken in dit component.

Note: ik zie nu dat deze al in een `.env` staan. Je kunt deze aanroepen via `process.env.GATSBY_ADMIN_URL` en dan kan je de variabele helemaal verwijderen: `fetch(`${process.env.GATSBY_ADMIN_URL}/applications/${id}`, {`

Reply:
Nu kan je ook de `useState` verwijderen: `const [context, setContext] = React.useState(null);`.